From 48773899ea431377d52c77a3c54355ff4c8ffa4f Mon Sep 17 00:00:00 2001
From: thrishalig <thrishalig@ami.com>
Date: Wed, 9 Aug 2023 22:29:44 +0530
Subject: [PATCH] checking Error completion code for DCMI set configuration 
 param 2  Option 60 And 43

Tested:

Before fix:

msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.100.111 -I lanplus -U root -P 0penBmc -C 17 raw 0x2c 0x12 0xdc 0x02 0x00 0x02
 dc

After fix:

msgubuntu4@msgubuntu4-OptiPlex-5080:~$ ipmitool -H 10.0.96.234 -I lanplus -U root -P 0penBmc -C 17 raw 0x2c 0x12 0xdc 0x02 0x00 0x02
Unable to send RAW command (channel=0x0 netfn=0x2c lun=0x0 cmd=0x12 rsp=0xd5): Command not supported in present state


Signed-off-by: thrishalig <thrishalig@ami.com>
---
 dcmihandler.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dcmihandler.cpp b/dcmihandler.cpp
index 73d225e..aa68286 100644
--- a/dcmihandler.cpp
+++ b/dcmihandler.cpp
@@ -1102,6 +1102,10 @@ ipmi::RspType<> setDCMIConfParams(uint8_t paramSelect,uint8_t setSelect, ipmi::m
                 {
                     return ipmi::responseCommandDisabled();
                 }
+		if(data & DCMI_OPTION_60_43_MASK)
+		{
+		    return ipmi::responseCommandNotAvailable();
+		}
                 if (data & DCMI_OPTION_12_MASK)
                 {
                     dcmi::setDHCPOption(DHCP_OPT12_ENABLED, true);
-- 
2.25.1

